### PR TITLE
Appropriately set status for router ports

### PR DESCRIPTION
**Pull request by snaiksat**
The router port is set to ACTIVE if the APIC config
for L3 succeeds.

In the case of external gateway setting, there is no
configuration on the APIC the port is always set to ACTIVE
once its created successfully in L3 DB code.

### CODE REVIEW

**Comment by amittbose** [comment]
I wonder if it is more appropriate to set the port status from the ML2 driver instead of L3. It'll definitely be more straightforward because all you need to do is added the update status calls to _perform_port_operations() and delete_port_postcommit().

**Comment by snaiksat** [comment]
This change of port status is specific to the router interface port. We want to be able to set this status only after the contract creation is successful on the APIC side, and which is where we are doing it in this patch.

File: apic_ml2/neutron/services/l3_router/l3_apic.py [hunk 1]
```diff
@@ -69,6 +73,24 @@ def _map_names(self, context, tenant_id, router, network, subnet):
                 ctx, subnet['id'])
         return atenant_id, arouter_id, anet_id, asubnet_id
 
+    def _get_port_id_for_router_interface(self, context, router_id, subnet_id):
+        filters = {'device_id': [router_id],
+                   'device_owner': ['network:router_interface'],
```
Comment (amittbose) [inline apic_ml2/neutron/services/l3_router/l3_apic.py]: Use qconst.DEVICE_OWNER_ROUTER_INTF and on line 88?

**Comment by snaiksat** [comment]
Good suggestion, done.

**Pull request merged**